c-api/src/element.rs

```diff
@@ -228,18 +228,22 @@ pub extern "C" fn lol_html_element_add_end_tag_handler(
 ) -> c_int {
     let element = to_ref_mut!(element);
 
-    match element.end_tag_handlers() {
-        Some(handlers) => {
-            handlers.push(Box::new(move |end_tag| {
-                match unsafe { handler(end_tag, user_data) } {
-                    RewriterDirective::Continue => Ok(()),
-                    RewriterDirective::Stop => Err("The rewriter has been stopped.".into()),
-                }
-            }));
-            0
+    unwrap_or_ret_err_code! {
+        match element.end_tag_handlers() {
+            Some(handlers) => {
+                handlers.push(Box::new(move |end_tag| {
+                    match unsafe { handler(end_tag, user_data) } {
+                        RewriterDirective::Continue => Ok(()),
+                        RewriterDirective::Stop => Err("The rewriter has been stopped.".into()),
+                    }
+                }));
+                Ok::<(), &'static str>(())
+            }
+            None => Err("No end tag.".into())
```

Comment:
Nit, we could simplify it a bit:

```rust
let handlers = unwrap_or_ret_err_code! { 
     element.end_tag_handlers().ok_or_else(|| "No end tag.".into()))
};

handlers.push(...);

0
```

Reply:
Thanks, that's much cleaner.

`ok_or_else(|| "No end tag.".into())` causes the following error:
```
error[E0282]: type annotations needed
   --> /home/harris/github/lol-html/c-api/src/element.rs:232:64
    |
232 |         element.end_tag_handlers().ok_or_else(|| "No end tag.".into())
    |                                                                ^^^^
    |
   ::: /home/harris/github/lol-html/c-api/src/lib.rs:67:75
    |
67  |                 crate::errors::LAST_ERROR.with(|e| *e.borrow_mut() = Some(err.into()));
    |                                                                           --- type must be known at this point
    |
help: try using a fully qualified path to specify the expected types
    |
232 |         element.end_tag_handlers().ok_or_else(|| <&str as Into<T>>::into("No end tag."))
    |                                                  ++++++++++++++++++++++++             ~

```
Without the `.into()`, everything builds and tests pass, so I'm going to go with that unless that's a mistake for some reason.